app/src/main/convert.js

```diff
@@ -37,7 +37,16 @@ export default async function (exportOptions) {
     }
   });
 
-  app.kap.mainWindow.send('export-progress', {text: ''});
+  ipcMain.on('cancel-export', () => {
+    convertProcess.cancel();
+  });
 
-  return outputPath;
+  return convertProcess
+    .then(() => {
+      app.kap.mainWindow.send('export-progress', {text: ''});
+      return outputPath;
+    })
+    .catch(err => {
```

Comment:
@sindresorhus how do you think I should handle the error? if this convert function errors (for example when process gets cancelled) the Kap plugin throws a giant error...

Reply:
You need to handle the `CancelError` here and ignore it, but rethrow if it's not a `CancelErro`: https://github.com/sindresorhus/p-cancelable#pcancelablecancelerror

Reply:
@sindresorhus the problem is, the user of this `convert` function needs a `outputPath` and if there is none, it will throw

Reply:
Then you need to handle the cancelation higher up where they expect the `outputPath`, or handle the cancelation here, and handle not getting a `outputPath` where something expects it.